Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,371 @@
+        }
+
+        CinemachineSplineRollExtension m_RollExtension; // don't use this directly
+        internal CinemachineSplineRollExtension rollExtension
+        {
+            private get
+            {
+                if (m_RollExtension == null)
+                    m_Spline.TryGetComponent(out m_RollExtension); // check if our spline has roll extension
+
+                return m_RollExtension;
+            }
+            set
+            {
+                m_RollExtension = value; // called by CinemachineSplineRollExtension
+            }
+        }
+        /// <summary>Positions the virtual camera according to the transposer rules.</summary>
+        /// <param name="curState">The current camera state</param>
+        /// <param name="deltaTime">Used for damping.  If less that 0, no damping is done.</param>
+        public override void MutateCameraState(ref CameraState curState, float deltaTime)
+        {
+            if (!IsValid)
+            {
+                m_SplinePosition = 0;
+                return;
+            }
+
+            var spline = m_Spline.Spline;
+            if (spline == null || spline.Count == 0)
+            {
+                return;
+            }
+            if (spline.Count == 1)
+            {
+                curState.RawPosition = m_PreviousCameraPosition = spline[0].Position;
+                curState.RawOrientation = m_PreviousOrientation = spline[0].Rotation;
+                return;
+            }
+            
+            // splines work with normalized position by default, so we convert m_SplinePosition to normalized at the start
+            var normalizedSplinePosition = 
+                SplineUtility.ConvertIndexUnit(spline, m_SplinePosition, m_PositionUnits, PathIndexUnit.Normalized);
+            
+            // Init previous frame state info
+            if (deltaTime < 0 || !VirtualCamera.PreviousStateIsValid)
+            {
+                m_PreviousNormalizedSplinePosition = normalizedSplinePosition;
+                m_PreviousCameraPosition = curState.RawPosition;
+                m_PreviousOrientation = curState.RawOrientation;
+            }
+
+            // Get the new ideal spline base position
+            if (m_AutoDolly.m_Enabled && FollowTarget != null)
+            {
+                // convert follow target into spline local space, because SplineUtility works in spline local space
+                SplineUtility.GetNearestPoint(spline, 
+                    m_Spline.transform.InverseTransformPoint(FollowTargetPosition), out _, out normalizedSplinePosition, 
+                    m_AutoDolly.m_SearchResolution, m_AutoDolly.m_SearchIteration);
+                m_SplinePosition = SplineUtility.ConvertIndexUnit(spline, normalizedSplinePosition, 
+                    PathIndexUnit.Normalized, m_PositionUnits);
+                
+                // Apply the spline position offset
+                normalizedSplinePosition += m_AutoDolly.m_PositionOffset;
```

Comment:
Is m_PositionOffset always in normalized units?  It should be using the same units as m_SplinePosition